### PR TITLE
Update `BA2025.EnableShadowStack` message

### PR DESCRIPTION
**Pull request by shaopeng-gh**
Update `BA2025.EnableShadowStack` message to include additional info.
The requirement is:

As far as BinSkim is concerned, the tool is doing the right thing, i.e., appears to have issued an accurate finding and did not emit it as an 'Error' (mandating a fix). We can definitely improve things, however, by updating our user-facing output/docs to note this special consideration. 'Older versions of .NET are not compatible with CET/shadow stack technology. If your native process loads older managed assemblies (.NET 6 or earlier), unhandled exceptions in those components may not be handled properly and may cause your process to crash').

We will take that fix immediate.

### CODE REVIEW

**Comment by shaopeng-gh** [comment]
>         /// <summary>

Currently the build fail is from `ComponentGovernanceComponentDetection` task,
not related to our changes.

```
##[error]There was an unexpected error: 
System.ArgumentException: An item with the same key has already been added. Key: Microsoft.TestPlatform.TestHost 17.4.1 - NuGet
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
   at Microsoft.ComponentDetection.Orchestrator.Experiments.Models.ExperimentDiff..ctor(IEnumerable`1 controlGroupComponents, IEnumerable`1 experimentGroupComponents)
   at Microsoft.ComponentDetection.Orchestrator.Experiments.ExperimentService.FinishAsync()
   at Microsoft.ComponentDetection.Orchestrator.Services.DetectorProcessingService.ProcessDetectorsAsync(IDetectionArguments detectionArguments, IEnumerable`1 detectors, DetectorRestrictions detectorRestrictions)
   at Microsoft.ComponentDetection.Orchestrator.Services.BcdeScanExecutionService.ExecuteScanAsync(IDetectionArguments detectionArguments)
   at Microsoft.ComponentDetection.Orchestrator.Services.BcdeScanCommandService.HandleAsync(IScanArguments arguments)
   at Microsoft.ComponentDetection.Internal.AzureDevOps.Services.BcdeWrapperService.ExecuteScanAsync(DetectionArguments detectionArguments) in /home/runner/work/component-detection-internal/component-detection-internal/ComponentDetection/src/Microsoft.ComponentDetection.Internal.Orchestrator/Services/BcdeWrapperService.cs:line 64
   at Microsoft.ComponentDetection.Internal.AzureDevOps.Services.ComponentDetectionCommandService.HandleAsync(IComponentDetectionArguments inputArguments) in /home/runner/work/component-detection-internal/component-detection-internal/ComponentDetection/src/Microsoft.ComponentDetection.Internal.Orchestrator/Services/ComponentDetectionCommandService.cs:line 111
For more information about configuring the build task please visit: [https://aka.ms/cgbuildtaskdocs](https://aka.ms/cgbuildtaskdocs)

```



---
Refers to: src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs:24 in 4f5c540. [](commit_id = 4f5c5408ceae319163a377bc88c5291241060bac, deletion_comment = False)

File: src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs [hunk 1]
```diff
@@ -25,6 +25,10 @@ public class EnableShadowStack : WindowsBinaryAndPdbSkimmerBase
         /// Control-flow Enforcement Technology (CET) Shadow Stack is a computer processor feature
         /// that provides capabilities to defend against return-oriented programming (ROP) based
         /// malware attacks.
+        /// Note: older versions of .NET are not compatible with CET/shadow stack technology.
```
Comment (suvamM) [inline src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs:28]: @shaopeng-gh This is really a note for the users right? Why add it as a comment here? It does not affect the analysis in any way correct?

File: src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs [hunk 2]
```diff
@@ -106,6 +110,10 @@ public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext conte
 
             // '{0}' does not enable the Control-flow Enforcement Technology (CET) Shadow Stack mitigation.
             // To resolve this issue, pass /CETCOMPAT on the linker command lines.
+            // Note: older versions of .NET are not compatible with CET/shadow stack technology.
```
Comment (suvamM) [inline src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs:113]: @shaopeng-gh Same comment here. Why are we adding this note here?

**Review by suvamM — APPROVED** [review]
I think a couple of comments can be removed. Other than that, the changes look good to me.

File: src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs (hunk 1 above)
Comment (shaopeng-gh) [reply src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs:28]: correct it does not affect the analysis in any way, our current pattern is to add these strings as comment in the code, so developer don't need to open the resource string and find what it is, easier to understand.
And we keep these comments in sync with the actual resource string.

Comment (suvamM) [reply src/BinSkim.Rules/PERules/BA2025.EnableShadowStack.cs:28]: Understood. All good then.

**Pull request merged**